File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/namespace_to_root/kotlin/com/example/kv/Kv.kt
package com.example.kv

import com.novi.bincode.BincodeDeserializer
import com.novi.bincode.BincodeSerializer
import com.novi.serde.DeserializationError
import com.novi.serde.Deserializer
import com.novi.serde.Serializer

data class Entry(
    val level: com.example.kv.Level,
    val outcome: com.example.kv.Outcome,
) {
    fun serialize(serializer: Serializer) {
        serializer.increase_container_depth()
        level.serialize(serializer)
        outcome.serialize(serializer)
        serializer.decrease_container_depth()
    }

    fun bincodeSerialize(): ByteArray {
        val serializer = BincodeSerializer()
        serialize(serializer)
        return serializer.get_bytes()
    }

    companion object {
        fun deserialize(deserializer: Deserializer): Entry {
            deserializer.increase_container_depth()
            val level = com.example.kv.Level.deserialize(deserializer)
            val outcome = com.example.kv.Outcome.deserialize(deserializer)
            deserializer.decrease_container_depth()
            return Entry(level, outcome)
        }

        @Throws(DeserializationError::class)
        fun bincodeDeserialize(input: ByteArray?): Entry {
            if (input == null) {
                throw DeserializationError("Cannot deserialize null array")
            }
            val deserializer = BincodeDeserializer(input)
            val value = deserialize(deserializer)
            if (deserializer.get_buffer_offset() < input.size) {
                throw DeserializationError("Some input bytes were not read")
            }
            return value
        }
    }
}

File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/namespace_to_root/csharp/Example/kv/Kv.cs
using CommunityToolkit.Mvvm.ComponentModel;
using Facet.Runtime.Serde;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Facet.Runtime.Bincode;

namespace Example.Kv;

public partial class Entry : ObservableObject, IFacetSerializable, IFacetDeserializable<Entry> {
    [ObservableProperty]
    private Example.Kv.Level _level;
    [ObservableProperty]
    private Example.Kv.Outcome _outcome;

    public void Serialize(ISerializer serializer)
    {
        serializer.IncreaseContainerDepth();
        Example.Kv.LevelBincode.Serialize(Level, serializer);
        Outcome.Serialize(serializer);
        serializer.DecreaseContainerDepth();
    }

    public static Entry Deserialize(IDeserializer deserializer)
    {
        deserializer.IncreaseContainerDepth();
        var level = Example.Kv.LevelBincode.Deserialize(deserializer);
        var outcome = Example.Kv.Outcome.Deserialize(deserializer);
        deserializer.DecreaseContainerDepth();
        return new Entry {
            Level = level,
            Outcome = outcome,
        };
    }

    public byte[] BincodeSerialize()
    {
        var serializer = new BincodeSerializer();
        Serialize(serializer);
        return serializer.GetBytes();
    }

    public static Entry BincodeDeserialize(byte[] input)
    {
        if (input is null)
        {
            throw new DeserializationError("Cannot deserialize null array");
        }
        var deserializer = new BincodeDeserializer(input);
        var value = Deserialize(deserializer);
        if (deserializer.GetBufferOffset() < input.Length)
        {
            throw new DeserializationError("Some input bytes were not read");
        }
        return value;
    }
}
